watertap/costing/unit_models/clarifier.py

```diff
@@ -0,0 +1,316 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import math
+import pyomo.environ as pyo
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.misc import StrEnum
+from ..util import (
+    make_capital_cost_var,
+    make_fixed_operating_cost_var,
+    register_costing_parameter_block,
+)
+
+
+class ClarifierType(StrEnum):
+    circular = "circular"
+    rectangular = "rectangular"
+    primary = "primary"
+
+
+def cost_clarifier(blk, clarifier_type=ClarifierType.circular, **kwargs):
+    """
+    Clarifier costing method
+
+    Args:
+        clarifier_type: ClarifierType Enum indicating clarifier type,
+            default = ClarifierType.circular
+
+        # `**kwargs`: Additional keywords for the ClarifierType, e.g., NaOCl
+        #     and CaOH2 mixers expect the `dosing_rate` keyword
+        #     argument.
+    """
+    if clarifier_type == clarifier_type.circular:
+        cost_circular_clarifier(blk, **kwargs)
+    elif clarifier_type == clarifier_type.rectangular:
+        cost_rectangular_clarifier(blk, **kwargs)
+    elif clarifier_type == clarifier_type.primary:
+        cost_primary_clarifier(blk, **kwargs)
+    else:
+        raise ConfigurationError(
+            f"{blk.unit_model.name} received invalid argument for clarifier_type:"
+            f" {clarifier_type}. Argument must be a member of the ClarifierType Enum."
+        )
+
+
+def build_circular_clarifier_cost_param_block(blk):
```

Comment:
I think it'd be good to note the refs too at this point.

Reply:
Yeah, good suggestion, I add reference at the begining and note them with respect to specific costing method.